<launch>
  <!-- send parameters for multidof -->
  <rosparam command="load" ns="robot_description_planning" file="$(find pr2_arm_navigation_config)/config/pr2_planning_description.yaml" />

  <node pkg="planning_environment" type="environment_server" output="screen" name="environment_server">      
    <param name="use_monitor" type="bool" value="true"/>
    <param name="use_collision_map" type="bool" value="true"/> <!-- set to false if problematic -->
  </node>

  <node pkg="planning_environment" type="planning_scene_validity_server" output="screen" name="planning_scene_validity_server"/>      
  
  <!-- load perception -->
  <include file="$(find pr2_arm_navigation_perception)/launch/laser-perception.launch"/>

  <!-- load move_arm -->
  <include file="$(find pr2_arm_navigation_actions)/launch/move_left_arm.launch"/>
  <include file="$(find pr2_arm_navigation_actions)/launch/move_right_arm.launch"/>

  <!-- load planning_environment -->

  <!-- OMPL -->
  <node pkg="ompl_ros_interface" type="ompl_ros" name="ompl_planning" output="screen">
    <param name="default_planner_config" type="string" value="SBLkConfig1"/>
    <rosparam command="load" file="$(find pr2_arm_navigation_planning)/config/right_left_arm_groups.yaml" />
    <rosparam command="load" file="$(find pr2_arm_navigation_planning)/config/ompl_planning.yaml" />
  </node>

  <node pkg="trajectory_filter_server" name="trajectory_shortcutting_filter_server" type="trajectory_filter_server" output="screen">
    <rosparam command="load" file="$(find move_arm)/joint_limits_right_left.yaml"/>
    <rosparam command="load" file="$(find move_arm_warehouse)/config/shortcutting_filters.yaml"/>
  </node>

  <!-- CHOMP -->
  <include file="$(find move_arm_warehouse)/launch/chomp_planning.launch"/>

  <node pkg="trajectory_filter_server" name="trajectory_smoothing_filter_server" type="trajectory_filter_server" output="screen">
    <rosparam command="load" file="$(find move_arm)/joint_limits_right_left.yaml"/>
    <rosparam command="load" file="$(find move_arm_warehouse)/config/smoothing_filters.yaml"/>
  </node>

  <!-- load ik -->
  <include file="$(find pr2_arm_navigation_kinematics)/launch/left_arm_collision_free_ik.launch"/>
  <include file="$(find pr2_arm_navigation_kinematics)/launch/right_arm_collision_free_ik.launch"/>

</launch>

